components/User/TopUserList.tsx

```diff
@@ -0,0 +1,157 @@
+import { Row, Col, Table, Form, Button, Image, Badge } from 'react-bootstrap';
+import { TopUser } from '../../models/User';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faEnvelope } from '@fortawesome/free-solid-svg-icons';
+
+import { faGithub } from '@fortawesome/free-brands-svg-icons';
+
+import styles from '../../styles/TopUserList.module.less';
+export interface TopUserListProps {
+  value: TopUser[];
+}
+
+const parseFn = (str: string) => (str ? JSON.parse(str) : {});
+export class TopUserList {
+  static Layout = ({ value = [] }: TopUserListProps) => (
+    <>
+      <Row className={styles.topUserRow}>
+        <Col sm={12} xs={12} lg={7}>
+          <div className={styles.showTitle}>
+            <div className={styles.showMedal}>
+              <i></i>
+              <i></i>
+              <i></i>
+            </div>
+            <h3 data-text="黑客馆" >黑客馆</h3>
+          </div>
+          <ul className={styles.topUserUl}>
+            {value.slice(0, 3).map(({ userId, score, rank, user }) => (
+              <li key={userId}>
+                <div>
+                  <div className={styles.imgBox}>
+                    <Image
+                      className={styles.topUserAvatar}
+                      src={user?.photo || parseFn(user?.oAuth)?.avatar_url}
+                      alt={
+                        user?.nickname ||
+                        user?.name ||
+                        user?.username ||
+                        '神秘黑客'
+                      }
+                    />
+                  </div>
+                  <div className={styles.showBox}>
+                    <i></i>
+                    <span>
+                      {user?.nickname ||
+                        user?.name ||
+                        user?.username ||
+                        '神秘黑客'}
+                    </span>
+                    <strong>{score}</strong>
+                    <p>
+                      {user?.email && (
+                        <a
+                          rel="noreferrer"
+                          target="_blank"
+                          href={'mailto:' + user.email}
+                        >
+                          <FontAwesomeIcon icon={faEnvelope} />
+                        </a>
+                      )}
+                      &nbsp;
+                      {parseFn(user?.oAuth)?.html_url && (
+                        <a
+                          rel="noreferrer"
+                          target="_blank"
+                          href={parseFn(user?.oAuth)?.html_url}
+                        >
+                          <FontAwesomeIcon icon={faGithub} />
+                        </a>
+                      )}
+                    </p>
```

Comment:
```suggestion
                    <address>
                      {user?.email && (
                        <a href={'mailto:' + user.email}>
                          <FontAwesomeIcon icon={faEnvelope} />
                        </a>
                      )}
                      {parseFn(user?.oAuth)?.html_url && (
                        <a
                          className="ms-2"
                          rel="noreferrer"
                          target="_blank"
                          href={parseFn(user?.oAuth)?.html_url}
                        >
                          <FontAwesomeIcon icon={faGithub} />
                        </a>
                      )}
                    </address>
```
1. 标签语义
2. 非页面链接浏览器会自动弹框，不会刷新当前页
3. 间距用 CSS 工具类：https://getbootstrap.com/docs/5.2/utilities/spacing/
4. 可以抽个函数组件给下面复用